tests/test_image.py

```diff
@@ -14,3 +16,41 @@ def test_from_url_with_non_OK_response(httpserver: HTTPServer):
 
     with raises(RequestException) as e:
         Image.from_url(httpserver.url_for(path))
+
+
+def test_from_image_source_on_local_file_path_pathlib():
+    image_source_path = Path(__file__).parent / "dog-and-cat-cover.jpg"
+    img_1 = ImagePrompt.from_image_source(image_source=image_source_path)
+    img_2 = ImagePrompt.from_file(path=image_source_path)
+    assert img_1.base_64 == img_2.base_64
+
+
+def test_from_image_source_on_local_file_path_str():
+    image_source_path = Path(__file__).parent / "dog-and-cat-cover.jpg"
+    image_source_path = str(image_source_path)
+    img_1 = ImagePrompt.from_image_source(image_source=image_source_path)
+    img_2 = ImagePrompt.from_file(path=image_source_path)
+    assert img_1.base_64 == img_2.base_64
+
+
+def test_from_image_source_on_url():
+    image_source_url = (
+        "https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png"
+    )
+    img_1 = ImagePrompt.from_image_source(image_source=image_source_url)
+    img_2 = ImagePrompt.from_url(url=image_source_url)
+    assert img_1.base_64 == img_2.base_64
+
+
+def test_from_image_source_on_bytes():
+    with open(Path(__file__).parent / "dog-and-cat-cover.jpg", "rb") as f:
```

Comment:
Can be replaced by `image_source_bytes = (Path(__file__).parent / "dog-and-cat-cover.jpg").read_bytes()`

Reply:
Replaced with a proposed solution